core/shared/src/main/scala/org/http4s/headers/Content-Encoding.scala

```diff
@@ -25,12 +25,17 @@ object `Content-Encoding` {
 
   private[http4s] val parser = ContentCoding.parser.map(`Content-Encoding`(_))
 
+  val name = ci"Content-Encoding"
+
   implicit val headerInstance: Header[`Content-Encoding`, Header.Single] =
     Header.createRendered(
-      ci"Content-Encoding",
+      name,
       _.contentCoding,
       parse,
     )
 }
 
-final case class `Content-Encoding`(contentCoding: ContentCoding)
+final case class `Content-Encoding`(contentCoding: ContentCoding) extends Header.AsRaw1 {
+  lazy val asRaw1: Header.Raw =
```

Comment:
What's preventing us from a generic implementation of this?  Access to the companion and the instance?

If we do this on the case classes, we'll be adding some bytes of overhead to every modeled header.  That's probably okay, but it's not free.  I was expecting to see this on cached values of particularly common header values.

I think rolling this out on some modeled headers, but not all, could be quite confusing.

Reply:
> Access to the companion and the instance?

Note that going in that direction somewhat gets you back to the old design which required path projections. I'm not at all opposed to experimentation here ofc, but worth calling that out

Reply:
@rossabaker
> I think rolling this out on some modeled headers, but not all, could be quite confusing.

Absolutely, I agree. All modelled headers need to be updated with this, hence the draft mode for this PR.

> If we do this on the case classes, we'll be adding some bytes of overhead to every modeled header. That's probably okay, but it's not free. I was expecting to see this on cached values of particularly common header values.

I'm not sure that's true. Every modelled header is converted into a `Header.Raw` anyway, this does does it pre-emptively and avoids the 12 allocations described in https://github.com/http4s/http4s/pull/6718#issuecomment-1257675532.

> What's preventing us from a generic implementation of this? Access to the companion and the instance?

Sorry not sure I understand what you mean by that. The point of having it as an interface and a lazy val is mostly useful for headers that are re-used often. I'd actually argue it's not very for headers like `Content-Header` as that changes for every request, but `Content-Encoding` probably doesn't change between requests (or ever, as in GZip middleware).